src/AiAdapter.js

```diff
@@ -496,15 +488,15 @@ var ticTacToe = ticTacToe || {};
    * Determines whether a square is empty.
    * @param {integer} square
    */
-  app.AiAdapter.prototype.squareIsEmpty = function (square) {
-    return (square === 0) ? true : false;
+  app.AiAdapter.prototype.isSquareEmpty = function (square, board) {
```

Comment:
Nope, Engine/Game will have a method to determine a legal move. A player doesn't need to have a method to determine if a square is empty, it will just be inefficient. 

For this model, think of the "game" as an adult teaching you to play.

Think about if you were a child playing the game for the first time and had no idea what the symbols on the board meant. You would try each spot to put your piece, the "game" would tell yes or no you cannot put your piece in a given spot. The board wouldn't stop you from doing it. Only once you have learned the meaning of symbols and rules of the game do you not need to check every spot. 

Therefore, our board should not care about what values occupy it's indices, only that is has the correct number of indices for the current game. The engine that is passing the board back and forth between players needs to care about such values as it should not allow an illegal move.